.github/workflows/ci.yml

```diff
@@ -0,0 +1,641 @@
+name: CI
+
+on:
+  push:
+    tags:
+      - '*'
+    paths-ignore:
+      - "**/*.md"
+  pull_request:
+    paths-ignore:
+      - "**/*.md"
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: write
+
+concurrency:
+  group: ci-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  ROOT: "${{ github.workspace }}"
+  SDK_PYTHON: "${{ github.workspace }}/sdk/python"
+  SCRIPTS: "${{ github.workspace }}/.github/scripts"
+  UV_PYTHON: "3.12"
+  PYODIDE_VERSION: "0.28.3"
+
+jobs:
+  # ============
+  # Python tests
+  # ============
+  python_tests:
+    name: Python ${{ matrix.python-version }} tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.10", "3.11", "3.12", "3.13" ]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v6
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run tests
+        shell: bash
+        working-directory: ${{ env.SDK_PYTHON }}
+        run: |
+          uv run --no-dev --group test pytest packages/flet/tests
+
+      - name: Run docs-coverage
+        if: matrix.python-version == '3.12'
+        shell: bash
+        working-directory: ${{ env.SDK_PYTHON }}
+        run: uv run --directory packages/flet --no-dev --group docs-coverage docstr-coverage > docstr_coverage.log 2>&1
+
+      - name: Upload docs-coverage logs
+        if: matrix.python-version == '3.12'
+        uses: actions/upload-artifact@v4
+        with:
+          name: docs-coverage
+          path: sdk/python/docstr_coverage.log
+
+  # ===========================
+  # Build Flet Flutter package
+  # ===========================
+  build_flet_package:
+    name: Build Flet Flutter package
+    runs-on: ubuntu-latest
+    outputs:
+      PKG_VER: ${{ steps.versions.outputs.PKG_VER }}
+      BUILD_VER: ${{ steps.versions.outputs.BUILD_VER }}
+      PYPI_VER: ${{ steps.versions.outputs.PYPI_VER }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0   # fetch all history
+          fetch-tags: true # ensure tags are available
+
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v6
+
+      - name: Compute versions
+        id: versions
+        run: source "${SCRIPTS}/update_build_version.sh"
+
+      - name: Setup Flutter
+        uses: kuhnroyal/flutter-fvm-config-action/setup@v3
+        with:
+          path: '.fvmrc'
+          cache: true
+
+      - name: Run tests
+        shell: bash
+        working-directory: packages/flet
+        run: flutter test
+
+      - name: Publish to pub.dev
+        shell: bash
+        env:
+          PUB_DEV_TOKEN: ${{ secrets.PUB_DEV_TOKEN }}
+        working-directory: packages/flet
+        run: |
+          if [[ "${GITHUB_REF_TYPE}" == "tag" ]]; then
```

Comment:
instead of bash `if` can we use `if:` in yaml and maybe split this job into two: `Publish to pub.dev` and `Publish to pub.dev dry run`?